apollo-ast/src/main/kotlin/com/apollographql/apollo3/ast/internal/ExecutableValidationScope.kt

```diff
@@ -310,6 +320,111 @@ internal class ExecutableValidationScope(
     }
   }
 
+  private fun GQLOperationDefinition.validateDeferDirectives() {
+    deferDirectivePathAndLabels.clear()
+    selectionSet.validateDeferDirectives("")
+  }
+
+  /**
+   * If a label is passed to a `@defer` directive, it must not be a variable, and it must be unique within all other `@defer` directives in
+   * the document.
+   *
+   * Also ensure that any `@defer` directive found when walking fragments on an operation have a unique path + label (Apollo-specific
+   * validation).
+   *
+   * For instance: this is invalid:
+   * ```
+   * query Query1 {
+   *   computers {
+   *   ...ComputerFields @defer  # path is computer
+   *   ...ComputerFields @defer  # path is computer
+   *   }
+   * }
+   * ```
+   *
+   * Also invalid:
+   * ```
+   * query Query2 {
+   *   computers {
+   *     ...ComputerFields
+   *     ...ComputerFields2
+   *     }
+   *   }
+   *
+   *   fragment ComputerFields on Computer {
+   *     screen {
+   *       ...ScreenFields @defer  # path is computer.screen
+   *     }
+   *   }
+   *
+   *   fragment ComputerFields2 on Computer {
+   *     screen {
+   *       ...ScreenFields @defer  # path is computer.screen
+   *     }
+   * }
+   * ```
+   */
+  private fun GQLSelectionSet.validateDeferDirectives(path: String) {
+    for (selection in selections) {
+      when (selection) {
+        is GQLField -> {
+          val fieldPath = if (path.isEmpty()) selection.name else "$path.${selection.name}"
+          selection.selectionSet?.validateDeferDirectives(fieldPath)
+        }
+        is GQLInlineFragment -> {
+          val deferDirective = selection.children.firstOrNull { it is GQLDirective && it.name == "defer" } as GQLDirective?
```

Comment:
There should be a `GQLInlineFragment.directives` property to avoid the use of the more generic `children`